# Splitting the data to k-folds
Link to the project: drinkability of water 
[(fr)](https://drive.google.com/file/d/1FGNR1O8EKGVKpVB_PMb5Ty2LipYgoM8q/view?usp=sharing)
[(kaggle)](https://www.kaggle.com/artimule/drinking-water-probability)

In this notebook, we will try to explore the new folds csv to see the number of samples per fold.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## Load the data

In [2]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
path = "/content/drive/MyDrive/Best ML model ever/input/dri_wat_pot_folds.csv"

df = pd.read_csv(path)
df = df.sample(frac=1, random_state=1).reset_index(drop=True)
df.head()

ph    Hardness       Solids  ...  Turbidity  Potability  kfold
0  7.273368  175.015083  14206.35732  ...   4.196421           1      0
1  6.977406  205.563500  16100.96795  ...   5.327585           0      1
2  8.195765  214.517610  10389.54254  ...   3.218709           1      2
3  7.080193  185.825726  39431.11265  ...   1.492207           1      4
4  7.371290  180.736526  25672.31200  ...   4.084697           0      4

[5 rows x 11 columns]

In [4]:
df['kfold'].value_counts()

0    656
3    655
1    655
4    655
2    655
Name: kfold, dtype: int64

All folds have 655 samples. This is expected as training data had 3276 value samples and we made 5 folds. So far, so good.

Now let's check the target distribution per fold.

In [9]:
df[df['kfold']==0]['Potability'].value_counts()

0    410
1    246
Name: Potability, dtype: int64

In [10]:
df[df['kfold']==1]['Potability'].value_counts()

0    406
1    249
Name: Potability, dtype: int64

In [11]:
df[df['kfold']==2]['Potability'].value_counts()

0    402
1    253
Name: Potability, dtype: int64

In [12]:
df[df['kfold']==3]['Potability'].value_counts()

0    379
1    276
Name: Potability, dtype: int64

In [13]:
df[df['kfold']==4]['Potability'].value_counts()

0    401
1    254
Name: Potability, dtype: int64

We see that in each fold, the distribution of targets is the same. This is what we
need. It can also be similar and doesn’t have to be the same all the time. Now, when
we build our models, we will have the same distribution of targets across every fold.